sdk/core/azure-core/src/main/java/com/azure/core/implementation/RestProxy.java

```diff
@@ -159,8 +159,7 @@ public Object invoke(Object proxy, final Method method, Object[] args) {
                 return handleHttpResponse(request, asyncDecodedResponse, methodParser, methodParser.getReturnType(),
                     context);
             }
-
-        } catch (Exception e) {
+        } catch (IOException e) {
```

Comment:
You're limiting the exception caught here to IOException from Exception. Is this what you meant to do?

Reply:
The error should catch as specific as possible. Based on the code, only unchecked IOException should catch right here. All other exception should handle inside of the methods. Did not see the point of catching general Exception.